core/llm/llms/Aider.ts

```diff
@@ -393,20 +379,33 @@ class Aider extends BaseLLM {
 
           this.aiderProcess.on("error", (error: Error) => {
             console.error(`Error starting Aider: ${error.message}`);
-            this.setAiderState("crashed");
+            
+            // Check if this is an authentication error for pearai_model
+            if (model === "pearai_model" && error.message.includes("authentication")) {
+              this.setAiderState("signedOut");  // Use new signedOut state
+            } else {
+              this.setAiderState("crashed");
+            }
+            
             clearTimeout(timeout);
             reject(error);
-            let message =
-              "PearAI Creator (Powered by aider) failed to start. Please contact PearAI support on Discord.";
+            
+            // Customize error message based on authentication state
+            let message = model === "pearai_model" && error.message.includes("authentication")
+              ? "Please sign in to use PearAI Creator"
```

Comment:
```suggestion
              ? "Please sign in to use PearAI Creator"
```
```suggestion
              ? "Please sign in to use PearAI Creator"
```
              ? "Please sign in to use PearAI Creator with hosted servers. You can also opt to use your own API-Key."